src/ucp/rndv/proto_rndv.c

```diff
@@ -48,17 +53,28 @@ ucp_proto_rndv_ctrl_reg_md_map(const ucp_proto_rndv_ctrl_init_params_t *params)
         /* Check the memory domain requires remote key, and capable of
          * registering the memory type
          */
-        md_index = ucp_proto_common_get_md_index(&params->super.super, lane);
-        md_attr  = &worker->context->tl_mds[md_index].attr;
+        ep_sys_dev = ucp_proto_common_get_sys_dev(&params->super.super, lane);
+        md_index   = ucp_proto_common_get_md_index(&params->super.super, lane);
+        md_attr    = &worker->context->tl_mds[md_index].attr;
         if (!(md_attr->cap.flags & UCT_MD_FLAG_NEED_RKEY) ||
             !(md_attr->cap.reg_mem_types & UCS_BIT(params->mem_info.type))) {
             continue;
         }
 
-        reg_md_map |= UCS_BIT(md_index);
-    }
+        *md_map |= UCS_BIT(md_index);
 
-    return reg_md_map;
+        if (ep_sys_dev < 64) {
+            *sys_dev_map |= UCS_BIT(ep_sys_dev);
+
+            mem_sys_dev = params->super.super.select_param->sys_dev;
```

Comment:
what would be the mem_sys_dev if memory comes from host memory? memory coming from non-host memory would have a sys device associated with them but not host memory right? Will we not take this path for host memory?

Reply:
it will be 0, we do not pack sys_dev info in case of SYS_DEVICE_UNKNOWN